File: backend/internal/domain/entity/order.go
package entity

import (
	"time"
)

type OrderRepository interface {
	CreateOrder(order *Order) (*Order, error)
	FindOrderById(id string) (*OrderComplete, error)
	FindAllOrders() ([]*OrderComplete, error)
	UpdateOrder(order *OrderComplete) (*Order, error)
	DeleteOrder(id string) error
}

type Order struct {
	ID          string    `json:"id" db:"id"`
	Priority    string    `json:"priority" db:"priority"`
	User_ID     string    `json:"user_id" db:"user_id"`
	Observation string    `json:"observation" db:"observation"`
	Status      string    `json:"status" db:"status"`
	Medicine_ID string    `json:"medicine_id" db:"medicine_id"`
	Quantity    int       `json:"quantity" db:"quantity"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
}

type OrderComplete struct {
	ID          string    `json:"id" db:"id"`
	Priority    string    `json:"priority" db:"priority"`
	Observation string    `json:"observation" db:"observation"`
	Status      string    `json:"status" db:"status"`
	Quantity    int       `json:"quantity" db:"quantity"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
	User        User      `json:"user"`
	Medicine    Medicine  `json:"medicine"`
	Medicine_ID string    `json:"medicine_id" db:"medicine_id"`
	User_ID     string    `json:"user_id" db:"user_id"`
}

func NewOrder(priority string, user_id string, observation string, medicine_id string, quantity int) *Order {
	return &Order{
		Priority:    priority,
		User_ID:     user_id,
		Observation: observation,
		Medicine_ID: medicine_id,
		Quantity:    quantity,
	}
}

File: mobile/lib/models/order.dart
import 'package:mobile/models/medicines.dart';
import 'package:mobile/models/user.dart';
class Order {
  String? createdAt;
  String? id;
  List<Medicines>? medicines;
  String? observation;
  String? priority;
  int? quantity;
  String? status;
  String? updatedAt;
  User? user;

  Order({
    this.createdAt,
    this.id,
    this.medicines,
    this.observation,
    this.priority,
    this.quantity,
    this.status,
    this.updatedAt,
    this.user,
  });

  Order.fromJson(Map<String, dynamic> json) {
    createdAt = json['created_at'];
    id = json['id'];
    medicines = <Medicines>[];
    if (json['medicine'] != null) {
      json['medicine'].forEach((x) {
        var medicine = Medicines.fromJson(x);
        if (medicines != null) {
          medicines!.add(medicine);
        }      
      });
    }
    observation = json['observation'];
    priority = json['priority'];
    quantity = json['quantity'];
    status = json['status'];
    updatedAt = json['updated_at'];
    user = json['user'] != null ? User.fromJson(json['user']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['created_at'] = createdAt;
    data['id'] = id;
    if (medicines != null) {
      data['medicine'] = medicines!.map((v) => v.toJson()).toList();
    }
    data['observation'] = observation;
    data['priority'] = priority;
    data['quantity'] = quantity;
    data['status'] = status;
    data['updated_at'] = updatedAt;
    if (user != null) {
      data['user'] = user!.toJson();
    }
    return data;
  }
}